src/librustc_mir/interpret/eval_context.rs

```diff
@@ -777,8 +776,13 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Our result will later be validated anyway, and there seems no good reason
         // to have to fail early here.  This is also more consistent with
         // `Memory::get_static_alloc` which has to use `const_eval_raw` to avoid cycles.
-        let val = self.tcx.const_eval_raw(param_env.and(gid))?;
-        self.raw_const_to_mplace(val)
+        let raw = self.tcx.const_eval_raw(param_env.and(gid))?;
```

Comment:
you can now revert all these changes as they are now unnecessary since the zst slice optimization is happening during promotion